contracts/RewardsLiquidator.vy

```diff
@@ -147,98 +175,147 @@ def _get_eth_anchor_price() -> uint256:
+            convert(block.timestamp, bytes32), #deadline
+            convert(eth_amount_in, bytes32),
+            convert(usdc_amount_out_min, bytes32),
+            convert(0, bytes32), #sqrtPriceLimitX96
         ),
         value=eth_amount_in,
-        max_outsize=128
+        max_outsize=32
     )
-    # The return type is uint256[] and the actual length of the array is 2, so the
-    # layout of the data is (pad32(64) . pad32(2) . pad32(arr[0]) . pad32(arr[1])).
-    # We need the second array item, thus the byte offset is 96.
-    return convert(extract32(result, 96), uint256)
-
+    return convert(result, uint256)
 
 @internal
 @pure
-def _get_min_amount_out(amount_in: uint256, price: uint256, max_diff_percent: uint256) -> uint256:
-    amount_out: uint256 = (amount_in * price) / 10**18
+def _get_min_amount_out(
+    amount: uint256,
+    price: uint256,
+    max_diff_percent: uint256,
+    decimal_token_in: uint256,
+    decimal_token_out: uint256
+) -> uint256:
+    amount_out: uint256 = (amount * price) / (10 ** decimal_token_in) # = (amount * (10 ** (18 - decimal_token_in)) * price) / 10 ** 18
     min_mult: uint256 = 10**18 - max_diff_percent
-    return (amount_out * min_mult) / 10**18
-
+    return (amount_out * min_mult) / (10 ** (36 - decimal_token_out)) # = ((amount_out * min_mult) / 10**18) / (10 ** (18 - decimal_token_out))
 
+# 1) stETH -> ETH (Curve)
+# 2) ETH -> USDC (Uniswap v2)
+# 3) USDC -> UST (Curve)
 @external
 def liquidate(ust_recipient: address) -> uint256:
     assert msg.sender == self.vault, "unauthorized"
 
     steth_amount: uint256 = ERC20(STETH_TOKEN).balanceOf(self)
     assert steth_amount > 0, "zero stETH balance"
 
-    steth_anchor_price: uint256 = self._get_steth_anchor_price()
-
+    # steth -> eth
+    steth_eth_price: uint256 = self._get_chainlink_price(CHAINLINK_STETH_ETH_FEED)
     min_eth_amount: uint256 = self._get_min_amount_out(
         steth_amount,
-        steth_anchor_price,
-        self.max_steth_price_difference_percent
+        steth_eth_price,
+        self.max_steth_eth_price_difference_percent,
+        STETH_TOKEN_DECIMALS,
+        WETH_TOKEN_DECIMALS
     )
 
     ERC20(STETH_TOKEN).approve(CURVE_STETH_POOL, steth_amount)
 
-    eth_amount: uint256 = CurveStableSwap(CURVE_STETH_POOL).exchange(
+    eth_amount: uint256 = CurvePool(CURVE_STETH_POOL).exchange(
         CURVE_STETH_INDEX,
         CURVE_ETH_INDEX,
         steth_amount,
         min_eth_amount
     )
-
     assert eth_amount >= min_eth_amount, "insuff. ETH return"
-    assert self.balance >= eth_amount
+    assert self.balance >= eth_amount, "ETH balance mismatch"
 
-    eth_anchor_price: uint256 = self._get_eth_anchor_price()
-
-    min_ust_amount: uint256 = self._get_min_amount_out(
+    # eth -> usdc
+    # assuming the USDC rate is almost the same as USD, due to the lack of the ETH/USDC feed
+    eth_usdc_price: uint256 = self._get_chainlink_price(CHAINLINK_ETH_USD_FEED)
```

Comment:
ETH/USDC price can be obtained using the USDC/USD feed used below

Reply:
fixed by obtaining exact rates for needed currencies from ChainLink and just calc reverse or cross rate if need. E.g. for USDC->UST:  1st: get USDC/ETH feed rate; 2nd: get UST/ETH feed rate and reverse it to ETH/UST; 3d: get cross USDC/UST rate by multiplying the previous rates.